### PR TITLE
fix(HtmlParser): close void elements on all node types

### LINKED ISSUE #5528: html parser: text nodes do not close void tags
`<h1> start <br> end </h1>`

br is not closed

### PR DESCRIPTION
**Pull request by vicb**
fixes #5528

### CODE REVIEW

**Comment by vicb** [comment]
Note: 
- Existing code would need to be updated `<ng-content></ng-content>` should become `<ng-content>`.
- follow-up: do we want to error when a void tag is closed ?

**Comment by vicb** [comment]
reviewed with Tobias

File: modules/angular2/src/compiler/html_tags.ts [hunk 1]
```diff
@@ -98,21 +101,20 @@ export class HtmlTagDefinition {
...
 var TAG_DEFINITIONS: {[key: string]: HtmlTagDefinition} = {
-  'link': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
-  'ng-content': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
-  'img': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
-  'input': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
-  'hr': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
-  'br': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
-  'wbr': new HtmlTagDefinition({closedByChildren: ['*'], closedByParent: true}),
+  'link': new HtmlTagDefinition({isVoid: true}),
+  'ng-content': new HtmlTagDefinition({isVoid: true}),
```
Comment (jelbourn) [inline modules/angular2/src/compiler/html_tags.ts]: `ng-content` can't be void; custom elements are not allowed to be so.

**Comment by jelbourn** [comment]
Closed via 9c6b929c7bd3d50223c3fe9ee298cef81a21493f and 62c2ed7c43595f8dae51eff9c9dd10f674d41a37

**Pull request closed without merging**